projects/step-core/src/lib/modules/basics/services/repository-parameters-storage.service.ts

```diff
@@ -0,0 +1,24 @@
+import { inject, Injectable } from '@angular/core';
+import { Observable, tap } from 'rxjs';
+import { PrivateApplicationService } from '../../../client/step-client-module';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RepositoryParametersStorageService {
+  private _api = inject(PrivateApplicationService);
+
+  private parametersInternal: ReadonlyArray<string> = [];
```

Comment:
you can also say ```= [] as const;```

Reply:
also I guess it makes more sense to say ```private parametersInternal: readonly string[] = []``` because when we are defining arrays we say T[] and not Array<T>

Reply:
@bojidaryovchev Yes I prefer `T[]` over `Array<T>`, because it is shorter, on the other hand `ReadonlyArray<T>` is more readable than `readonly T[]` in my opinion. Especially if we could have fields like `readonly someItems: readonly Item[] = []`, such notation looks too complicated in comparison with `readonly someItems: ReadonlyArray<Item> = []`.

Reply:
well not really, you don't have to remove the ```string[]``` part

Reply:
Can't agree
`let test: string[] = [] as const;` is invalid record
https://www.typescriptlang.org/play?#code/DYUwLgBGIM5gXBOAnAlgOwOYG0C6EBeCPCAQxggGMB7dOAbgCho4A6ABwFcYALACgDkAM2rUBASiY061UK2DVMfFmElA
Typescript shows an error `The type 'readonly []' is 'readonly' and cannot be assigned to the mutable type 'string[]'.`